framework/src/base/FlagElementsThread.C

```diff
@@ -24,11 +24,10 @@
 
 FlagElementsThread::FlagElementsThread(FEProblem & fe_problem,
                                        std::vector<Number> & serialized_solution,
-                                       DisplacedProblem * displaced_problem,
                                        unsigned int max_h_level) :
     ThreadedElementLoop<ConstElemRange>(fe_problem, fe_problem.getAuxiliarySystem()),
     _fe_problem(fe_problem),
-    _displaced_problem(displaced_problem),
+    _displaced_problem(_fe_problem.getDisplacedProblem()),
```

Comment:
To be clear: I'm being serious (After I made that comment I thought you might think I was being sarcastic!).  It is weird that we were passing in `displaced_problem` directly here....